**Цель работы:**

Осуществить предварительную обработку данных csv-файла, выявить и устранить проблемы в этих данных.

**Задание**
1. Загрузить датасет с помощью библиотеки ```pandas```.
2. Вывести первые 20 строк с помощью метода ```head```.
3. Выполнить обзор данных - описать столбцы и вашу предметную область.
4. С помощью метода ```info``` оценить данные (есть ли пропуски, сколько всего строк, какие типы данных у столбцов).
5. Применить ```describe```. С помощью ```describe``` оценить числовые столбцы (если они есть).
6. Вывести на экран названия столбцов с помощью ```df.columns```. Выявить проблемы с названиями, если они есть.
   При необходимости переименовать столбцы. Если проблемы не обнаружены также дать пояснения.
7. Проверить данные на наличие пропусков и устранить их, если они есть (пропуски
   необходимо либо удалить, либо заменить каким-то значением). Не забудьте пояснения!
8. Проверьте данные на наличие явных и неявных дубликатов. Удалите дубликаты, если они есть.
10. Проверьте типы данных, при необходимости измените типы данных, чтобы они соответствовали действительности.
11. Осуществите группировки и создайте сводные таблицы в соответствии с вариантом.
12. Сделайте выводе по работе.


# Загрузка набора данных

### Описание предметной области

    
**Предметная область**\
Данные отражают информацию о заёмщиках и их кредитах: возраст, доход, тип жилья, стаж работы, цель кредита, сумму займа, процентную ставку, длину кредитной истории и факт дефолта (наличие или отсутствие просрочки по выплатам). Анализ таких данных позволяет банкам и финансовым организациям оценивать надёжность клиентов, выявлять факторы, влияющие на риск невозврата кредита, и разрабатывать стратегии управления рисками.

**Вариант № 7**

Набор данных: ```credit_risk.csv```

Атрибуты:\
```Id``` — идентификатор\
```Age``` — возраст\
```Income```— доход\
```Home``` — статус домовладения (Собственный, Ипотека, Аренда)\
```Emp_length``` — стаж работы в годах\
```Intent``` — цель кредита\
```Amount``` — сумма кредита по заявке\
```Rate``` — процентная ставка\
```Status``` — статус одобрения кредита\
```Percent_income``` — сумма кредита в процентах от дохода\
```Default``` — не допустил ли заявитель ранее дефолта по кредиту (да, нет)\
```Cred_length``` — длина кредитной истории заявителя\

### 1.Чтение файла (набора данных)

Импорт библиотек, чтение файла с помощью ```pandas```

In [41]:
import pandas as pd
df = pd.read_csv("credit_risk.csv")

### 2. Обзор данных

**2.1 Вывод первых 20 строк с помощью метода ```head```.**

In [42]:
df.head(20)

Id   Age  Income      Home  Emp_length             Intent  Amount   Rate  \
0    0  22.0   59000      RENT       123.0           PERSONAL   35000  16.02   
1    1  21.0    9600       OWN         5.0          EDUCATION    1000  11.14   
2    2  25.0    9600  MORTGAGE         1.0            MEDICAL    5500  12.87   
3    3  23.0   65500      RENT         4.0            MEDICAL   35000  15.23   
4    4  24.0   54400      RENT         8.0            MEDICAL   35000  14.27   
5    5  21.0    9900       OWN         2.0            VENTURE    2500   7.14   
6    6  26.0   77100     RENET         8.0          EDUCATION   35000  12.42   
7    7  24.0   78956      RENT         5.0            MEDICAL   35000  11.11   
8    8  24.0   83000      RENT         8.0           PERSONAL   35000   8.90   
9    9  21.0   10000       OWN         6.0            VENTURE    1600  14.74   
10  10  22.0   85000      RENT         6.0            VENTURE   35000  10.37   
11  11  21.0   10000       OWN         2.0    HOMEIMPROVEMENT    4500   8.63   
12  12  23.0   95000      RENT         2.0            VENTURE   35000   7.90   
13  13  26.0  108160      RENT         4.0          EDUCATION   35000  18.39   
14  14  23.0  115000      RENT         2.0          EDUCATION   35000   7.90   
15  15  23.0  500000  MORTGAGE         7.0  DEBTCONSOLIDATION   30000  10.65   
16  16  23.0  120000      RENT         0.0          EDUCATION   35000   7.90   
17  17  23.0   92111      RENT         7.0            MEDICAL   35000  20.25   
18  18  23.0  113000      RENT         8.0  DEBTCONSOLIDATION   35000  18.25   
19  19  24.0   10800  MORTGAGE         8.0          EDUCATION    1750  10.99   

    Status  Percent_income Default  Cred_length  
0        1            0.59       Y            3  
1        0            0.10       N            2  
2        1            0.57       N            3  
3        1            0.53       N            2  
4        1            0.55       Y            4  
5        1            0.25       N            2  
6        1            0.45       N            3  
7        1            0.44       N            4  
8        1            0.42       N            2  
9        1            0.16       N            3  
10       1            0.41       N            4  
11       1            0.45       N            2  
12       1            0.37       N            2  
13       1            0.32       N            4  
14       0            0.30       N            4  
15       0            0.06       N            3  
16       0            0.29       N            4  
17       1            0.32       N            4  
18       1            0.31       N            4  
19       1            0.16       N            2

**2.2 Оценка данных с помощью метода ```info```**.

In [43]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 652 entries, 0 to 651
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Id              652 non-null    int64  
 1   Age             652 non-null    float64
 2   Income          652 non-null    int64  
 3   Home            652 non-null    object 
 4   Emp_length      643 non-null    float64
 5   Intent          652 non-null    object 
 6   Amount          652 non-null    int64  
 7   Rate            586 non-null    float64
 8   Status          652 non-null    int64  
 9   Percent_income  652 non-null    float64
 10  Default         652 non-null    object 
 11  Cred_length     652 non-null    int64  
dtypes: float64(4), int64(5), object(3)
memory usage: 61.3+ KB


---

**Пояснение:**\
В наборе данных 652 строки и 12 столбцов;\
Есть числовые и категориальные признаки;\
Пропуски обнаружены в столбцах ```Emp_length``` (9 штук) и ```Rate``` (66 штук);\
Эти пропуски нужно будет обработать (например, заполнить медианой).

---

**2.3 Оценка данных с помощью метода ```describe```.**

In [44]:
df.describe()

Id         Age         Income  Emp_length        Amount  \
count  652.000000  652.000000     652.000000  643.000000    652.000000   
mean   325.495399   24.285276   90008.855828    4.601866  18800.996933   
std    188.352535    7.841061   69530.312525    7.281616   9152.896250   
min      0.000000   21.000000    9600.000000    0.000000   1000.000000   
25%    162.750000   23.000000   44000.000000    2.000000  10000.000000   
50%    325.500000   24.000000   69998.000000    4.000000  21850.000000   
75%    488.250000   25.000000  128499.000000    7.000000  25000.000000   
max    649.000000  144.000000  500000.000000  123.000000  35000.000000   

             Rate      Status  Percent_income  Cred_length  
count  586.000000  652.000000      652.000000   652.000000  
mean    12.289471    0.601227        0.279831     3.012270  
std      3.272772    0.490022        0.150231     0.808843  
min      5.420000    0.000000        0.010000     2.000000  
25%     10.250000    0.000000        0.160000     2.000000  
50%     12.180000    1.000000        0.280000     3.000000  
75%     14.720000    1.000000        0.380000     4.000000  
max     21.210000    1.000000        0.830000     4.000000


---

**Пояснение:**\
Была произведена оценка числовых столбцов с помощью ```describe```\
В датасете 652 записей;\
Возраст в среднем 24 года, но есть аномалия: минимальный 21, максимальный 144;\
Доход сильно варьируется: от 9 600 до 500 000, средний около 90 000;\
Стаж работы (```emp_length```): медиана 4 года, но есть подозрительные значения (123 года);\
Сумма кредита (```amount```): от 1 000 до 35 000, медиана ~21 850;\
Ставка (```rate```): от 5.42% до 21.21%, среднее ~12.3%;\
Доля кредита от дохода (```percent_income```): в среднем 28%;\
Длина кредитной истории (```cred_length```): от 2 до 4 лет, медиана 3 года.

 ---


**2.4 Оценка названий столбцов**

In [45]:
df.columns

Index(['Id', 'Age', 'Income', 'Home', 'Emp_length', 'Intent', 'Amount', 'Rate',
       'Status', 'Percent_income', 'Default', 'Cred_length'],
      dtype='object')

In [46]:
df.columns = df.columns.str.lower()
df.columns

Index(['id', 'age', 'income', 'home', 'emp_length', 'intent', 'amount', 'rate',
       'status', 'percent_income', 'default', 'cred_length'],
      dtype='object')

---

**Пояснение:**\
Были выведеныф на экран названия столбцов. Не было обнаружено проблем.\
Названия столбцов были приведены к нижнему регистру для единообразия и удобства работы в коде,\
чтобы избежать ошибок при обращении к столбцам

 ---


### 3. Проверка пропусков

In [47]:
df.isna().sum()
df['emp_length'] = df['emp_length'].fillna(df['emp_length'].median())
df['rate'] = df['rate'].fillna(df['rate'].median())
df.isna().sum()

id                0
age               0
income            0
home              0
emp_length        0
intent            0
amount            0
rate              0
status            0
percent_income    0
default           0
cred_length       0
dtype: int64


---

**Пояснение:**\
Если удалить строки:\
Удалим 9 строк с ```emp_length``` - это 1,4% данных (не страшно).\
Удалим 66 строк с ```rate``` - это уже 10% данных (довольно много).\
Вывод: потеря в сумме 75 строк (11–12% данных).\
Т. е. уменьшается размер выборки, информация теряется.

Если заменить медианой\
```emp_length```: заменить медианой (например, 4 года) — получится реалистично.\
```rate```: заменить медианой (примерно 12.2%) — логично, потому что медиана отражает «типичное значение» и не портит статистику.\
Вывод: сохранение всех 652 строк, соответсвенно, анализ более устойчивый.

Пропуски в колонках ```emp_length``` (9 значений) и ```rate``` (66 значений) были заменены на медиану. Это позволило сохранить все строки датасета и использовать типичные значения вместо удаления данных.

 ---

### 4. Проверка дубликатов

#### Проверка явных дубликатов

In [48]:
df[df.duplicated()]

id   age  income  home  emp_length             intent  amount  rate  \
650  649  23.0   58800  RENT         7.0  DEBTCONSOLIDATION   20000  7.66   
651  649  23.0   58800  RENT         7.0  DEBTCONSOLIDATION   20000  7.66   

     status  percent_income default  cred_length  
650       1            0.34       N            3  
651       1            0.34       N            3

In [49]:
print(df.duplicated().sum())

2


In [50]:
df = df.drop_duplicates().reset_index(drop = True)
print(df.duplicated().sum())

0


---

**Пояснение:**\
```df.duplicated()```\
Эта команда проверяет, есть ли дубликаты строк в таблице.
Возвращает колонку из True и False:
- True - строка является дубликатом (повторяется),
- False - уникальная строка.

```print(df.duplicated().sum())```\
Команда выводит на экран количество строк-дубликатов

```df = df.drop_duplicates().reset_index(drop = True)```\
Команда удаляет дубликаты и обновляет индексы строк (чтобы потом при обращении к строкам не было «дырок» в номерах).

---

#### Проверка неявных дубликатов

In [51]:
unique_c = ['age', 'home', 'emp_length', 'intent', 'status', 'default', 'cred_length']
for i in unique_c:
  print(i,'\n', df[i].unique())

age 
 [ 22.  21.  25.  23.  24.  26. 144. 123.]
home 
 ['RENT' 'OWN' 'MORTGAGE' 'RENET' 'OTHER']
emp_length 
 [123.   5.   1.   4.   8.   2.   6.   7.   0.   9.   3.  10.  11.]
intent 
 ['PERSONAL' 'EDUCATION' 'MEDICAL' 'VENTURE' 'HOMEIMPROVEMENT'
 'DEBTCONSOLIDATION']
status 
 [1 0]
default 
 ['Y' 'N' 'No']
cred_length 
 [3 2 4]


In [52]:
for i in unique_c:
  print(df[i].value_counts())

age
24.0     137
23.0     130
25.0     130
22.0     109
26.0      98
21.0      43
144.0      2
123.0      1
Name: count, dtype: int64
home
RENT        413
MORTGAGE    136
OWN          98
OTHER         2
RENET         1
Name: count, dtype: int64
emp_length
0.0      91
2.0      76
3.0      70
5.0      64
6.0      61
1.0      60
7.0      56
4.0      55
8.0      46
9.0      45
10.0     22
123.0     2
11.0      2
Name: count, dtype: int64
intent
EDUCATION            145
VENTURE              117
PERSONAL             111
MEDICAL              107
DEBTCONSOLIDATION    102
HOMEIMPROVEMENT       68
Name: count, dtype: int64
status
1    390
0    260
Name: count, dtype: int64
default
N     497
Y     150
No      3
Name: count, dtype: int64
cred_length
3    224
4    217
2    209
Name: count, dtype: int64


In [53]:
# исправление возраста (замену на медиану возрастов)
median_age = df.loc[df['age'] < 100, 'age'].median()  # считаем медиану по "нормальным" возрастам
df.loc[df['age'] > 100, 'age'] = median_age
# исправление типа жильяя
df['home'] = df['home'].replace('RENET', 'RENT')
# исправление аномалии стажа работы (замена на медиану)
median_emp = df.loc[df['emp_length'] < 50, 'emp_length'].median()  # считаем медиану по адекватным значениям
df.loc[df['emp_length'] > 50, 'emp_length'] = median_emp
# округление дробного значения стажа возраста
df['emp_length'] = df['emp_length'].round()
# исправление категории дефолта
df['default'] = df['default'].replace('No', 'N')
# проверка результата 
print(df['age'].unique())
print(df['home'].unique())
print(df['emp_length'].unique())
print(df['default'].unique())

[22. 21. 25. 23. 24. 26.]
['RENT' 'OWN' 'MORTGAGE' 'OTHER']
[ 4.  5.  1.  8.  2.  6.  7.  0.  9.  3. 10. 11.]
['Y' 'N']


---

**Пояснение:**\
В данных обнаружены ошибки и неявные дубликаты:\
```age``` - нереалистичные значения (123, 144);\
```home``` - опечатка "RENET";\
```emp_length``` - ошибка (123 года стажа) и дробное значение 4.6;\
```default``` - смешение категорий (No и N);\
Эти проблемы нужно исправить на этапе предобработки, чтобы анализ был корректным.

Все аномалии и опечатки исправлены.\
Данные приведены к корректному виду, теперь с ними можно уверенно работать дальше: строить группировки, сводные таблицы и анализировать.


 ---

### 5. Провека типов данных

In [54]:
# Проверьте типы данных, при необходимости измените типы данных, чтобы они соответствовали действительности.
df.dtypes

# возраст и стаж работы должны быть целыми числами
df['age'] = df['age'].astype(int)
df['emp_length'] = df['emp_length'].astype(int)

# категориальные переменные
df['home'] = df['home'].astype('category')
df['intent'] = df['intent'].astype('category')
df['default'] = df['default'].astype('category')
df['status'] = df['status'].astype('category')

print(df.dtypes)

id                   int64
age                  int64
income               int64
home              category
emp_length           int64
intent            category
amount               int64
rate               float64
status            category
percent_income     float64
default           category
cred_length          int64
dtype: object


---

**Пояснение:**\
```id``` - ```int64```\
Идентификатор записи: тип целое число ```int64``` - верно.

```age``` - ```float64```\
Возраст должен быть целым числом, но хранится как ```float64```.

```income``` - ```int64```\
Доход (число): тип целое число ```int64``` - верно.

```home``` - ```object```\
Тип жилья это строка ```object```, но по смыслу это категориальная переменная, лучше ```category```.

```emp_length``` - ```float64```\
Стаж работы должен быть целым числом, лучше перевести в ```int64```.

```intent``` - ```object```\
Цель кредита сейчас строка, но это категориальная переменная - нужно перевести в ```category```.

```amount``` - ```int64```\
Сумма кредита: тип целое число ```int64``` - верно.

```rate``` - ```float64```\
Процентная ставка может иметь дробь - ```float64``` - верно.

```status``` - ```int64```\
Статус (0 или 1) сейчас как число, но по смыслу это категориальный/булевый признак - нужно ```category```.

```percent_income``` - ```float64```\
Доля дохода, идущая на кредит - дробное число - ```float64``` - верно.

```default``` - ```object```\
Дефолт (Y, N) сейчас строка, но это категориальная переменная - нужно перевести в ```category```.

```cred_length``` - ```int64```\
Длина кредитной истории (годы): целое число - тип ```int64``` - верно.

Проверка показала, что некоторые признаки определены неверно:\
```age``` и ```emp_length``` были записаны как ```float64```, хотя должны быть целыми числами;\
```home```, ```intent```, ```default``` и ```status``` определены как ```object или ```int64```, хотя по смыслу являются категориальными.\
Типы данных были приведены в соответствие.


 ---

### 6. Группировка данных

#### Задание 1


`Группировка - цель кредита (intent) по количеству статуса
домовладения (home)`

In [55]:
print(df.groupby(['intent', 'home'], observed=True)['id'].count())

intent             home    
DEBTCONSOLIDATION  MORTGAGE    21
                   OWN          9
                   RENT        72
EDUCATION          MORTGAGE    34
                   OTHER        1
                   OWN         16
                   RENT        94
HOMEIMPROVEMENT    MORTGAGE    13
                   OWN         16
                   RENT        39
MEDICAL            MORTGAGE    22
                   OWN         15
                   RENT        70
PERSONAL           MORTGAGE    19
                   OTHER        1
                   OWN         22
                   RENT        69
VENTURE            MORTGAGE    27
                   OWN         20
                   RENT        70
Name: id, dtype: int64


---

**Пояснение:**\
Была проведена группировка клиентов по цели кредита и типу жилья. Большинство клиентов, независимо от цели кредита,\
живут в аренде ```RENT```. На втором месте — клиенты с ипотекой ```MORTGAGE```.\
Владельцы собственного жилья ```OWN``` составляют меньшинство.\
Таким образом, арендаторы являются самой кредитозависимой группой

Более наглядный результат был сделан через ```pivot_table```

 ---

In [56]:
pivot = df.pivot_table(
    # строки = цель кредита
    index='intent',  
    # столбцы = тип жилья
    columns='home',
    # что считаем (количество клиентов)
    values='id',  
    # функция подсчёта
    aggfunc='count', 
    # показывать только реально существующие комбинации
    observed=True       
)

print(pivot)

home               MORTGAGE  OTHER   OWN  RENT
intent                                        
DEBTCONSOLIDATION      21.0    NaN   9.0  72.0
EDUCATION              34.0    1.0  16.0  94.0
HOMEIMPROVEMENT        13.0    NaN  16.0  39.0
MEDICAL                22.0    NaN  15.0  70.0
PERSONAL               19.0    1.0  22.0  69.0
VENTURE                27.0    NaN  20.0  70.0


#### Задание 2

`Группировка - статус домовладения (home) по количеству дефолтов (default). Создать датафрейм. Переименовать столбец с количеством в “сount”. Отсортировать по убыванию столбца “count”.`

In [57]:
# Посчитать количество строк в каждой группе (группа = комбинация home + default)
counts = df.groupby(['home', 'default'], observed = True).size()
# Превратить результат в DataFrame
df_group = counts.reset_index()
# Переименовать колонку с количеством в "count"
df_group.columns = ['home', 'default', 'count']
# Отсортировать по убыванию столбца count
df_group = df_group.sort_values(by='count', ascending=False)
# Посмотреть результат
print(df_group)

       home default  count
5      RENT       N    308
0  MORTGAGE       N    107
6      RENT       Y    106
3       OWN       N     85
1  MORTGAGE       Y     29
4       OWN       Y     13
2     OTHER       Y      2


---

**Пояснение:**\
Анализ распределения клиентов по типу жилья и факту дефолта показал, что арендаторы ```RENT``` составляют основную долю заёмщиков - 414 человек,\
из них 106 допустили дефолт. Среди ипотечных заёмщиков ```MORTGAGE``` наблюдается 29 дефолтов из 136 клиентов.\
Владельцы собственного жилья ```OWN``` составляют меньшую долю выборки и демонстрируют наименьший уровень дефолтов — всего 13 случаев.\
Таким образом, арендаторы являются наиболее рискованной категорией клиентов, а владельцы жилья — наименее рискованной

 ---

#### Задание 3

`Сводная таблица (pivot_table) - средняя сумма кредита amount для каждой цели кредита. Отсортировать по убыванию. Округлить до трёх знаков.`

In [58]:
pivot_amount = df.pivot_table(
    # строки = цель кредита
    index='intent',       
    # что считаем = сумма кредита
    values='amount',
    # функция = среднее значение
    aggfunc='mean',
    # показывать только то, что реально есть
    observed=True
)

# округление до 3 знаков
pivot_amount = pivot_amount.round(3)

# сортировка по убыванию
pivot_amount = pivot_amount.sort_values(by='amount', ascending=False)

print(pivot_amount)


                      amount
intent                      
DEBTCONSOLIDATION  19578.186
PERSONAL           19350.676
EDUCATION          19159.483
MEDICAL            18732.243
HOMEIMPROVEMENT    17884.559
VENTURE            17732.692


---

**Пояснение:**\
Наибольшие кредиты берутся для консолидации долгов, личных целей и образования (в среднем около 19 тыс.).\
На медицинские расходы и ремонт жилья кредиты несколько меньше (около 18 тыс.).\
Наименьшие кредиты наблюдаются в категории бизнес-целей ```VENTURE``` — около 17,7 тыс.\
Таким образом, консолидация долгов и личные кредиты являются наиболее «дорогими» целями для клиентов

 ---

#### Задание 4

`Сводная таблица (pivot_table) - средний доход income для каждого статуса домовладения (home) - строки и цели кредита - столбцы . Отсортировать по возрастанию home. Округлить до двух знаков.`

In [59]:
pivot_income = df.pivot_table(
    # строки = тип жилья
    index='home', 
    # столбцы = цель кредита
    columns='intent', 
    # значения = доход
    values='income',
    # функция = среднее
    aggfunc='mean',  
    # показывать только реальные комбинации
    observed=True        
)

# округление до 2 знаков
pivot_income = pivot_income.round(2)

# сортировка по возрастанию home (по алфавиту)
pivot_income = pivot_income.sort_index(ascending=True)

print(pivot_income)


intent    DEBTCONSOLIDATION  EDUCATION  HOMEIMPROVEMENT    MEDICAL   PERSONAL  \
home                                                                            
MORTGAGE          128852.05  141393.41        144012.62  117061.55  102129.53   
OTHER                   NaN   11389.00              NaN        NaN   19200.00   
OWN                14334.67   63824.75         72565.44   39385.07   69722.55   
RENT               85176.99   87780.28         83998.54   79992.63   86012.55   

intent      VENTURE  
home                 
MORTGAGE  145813.37  
OTHER           NaN  
OWN        44181.00  
RENT       88363.53  


---

**Пояснение:**\
Ипотечники ```MORTGAGE``` - самые богатые клиенты (средние доходы свыше 100 000).\
Владельцы жилья ```OWN``` - доходы средние (около 40 000–70 000).\
Арендаторы ```RENT``` - доходы тоже приличные (80 000–88 000), но ниже, чем у ипотечников.\
```OTHER``` - слишком мало данных, выводы неинформативны.

Сводная таблица показала, что наибольший средний доход наблюдается у клиентов с ипотекой ```MORTGAGE```,\
что подтверждает их более высокую платёжеспособность. Владельцы собственного жилья ```OWN``` имеют доходы ниже,\
а арендаторы ```RENT``` находятся на промежуточном уровне.\
Категория ```OTHER``` встречается крайне редко и статистически значимых выводов не даёт.


 ---

### Вывод

В ходе работы был проведён предварительный анализ набора данных ```credit_risk.csv```, содержащего информацию о 652 клиентах и их кредитах. Данные относятся к предметной области оценки кредитных рисков и включают показатели возраста, дохода, цели кредита, типа жилья, суммы займа, процентной ставки, длины кредитной истории и факта дефолта.
На этапе предобработки были выявлены и устранены ошибки: некорректные значения возраста (123 и 144 года) и стажа работы (123 года, дробное значение 4.6), опечатка в типе жилья («RENET» заменена на «RENT»), а также неявный дубликат в колонке дефолта («No» объединено с «N»). Пропуски в числовых данных были заполнены медианными значениями. Типы данных приведены в соответствие: числовые признаки переведены в ```int64``` или ```float64```, категориальные признаки — в ```category```.

Анализ группировок показал, что наибольшее количество клиентов берут кредиты, находясь в аренде жилья ```RENT```. Среди них же наблюдается и наибольшее количество дефолтов (106 случаев). В то же время владельцы собственного жилья ```OWN``` составляют меньшинство выборки и демонстрируют наименьший уровень дефолтов (13 случаев). Ипотечники ```MORTGAGE``` занимают промежуточное положение. Таким образом, арендаторы являются наиболее уязвимой категорией заёмщиков.

Построенные сводные таблицы показали, что самые крупные кредиты берутся для консолидации долгов, личных целей и образования (в среднем около 19 тыс.), тогда как кредиты на бизнес и ремонт жилья меньше (около 17–18 тыс.). Анализ доходов клиентов продемонстрировал, что ипотечники обладают наибольшим средним доходом (свыше 100 тыс.), арендаторы занимают промежуточное положение (около 85–88 тыс.), а владельцы жилья имеют более низкий доход (40–70 тыс.).

В целом проведённое исследование показало, что структура кредитного риска существенно зависит от типа жилья и цели кредита. Наиболее надёжными клиентами являются владельцы жилья, тогда как арендаторы чаще обращаются за кредитами и чаще уходят в дефолт.

В лабораторной работе осуществлялась работа с библиотекой ```pandas```. Были изучены типы данных в библиотеке, индексация и др. На практике были произведены подготовка данных, устранение ошибок и дубликатов, обработка различных типов данных. Были изучены такие понятия как группировка и категоризация.

### Дополнительное задание

**`Подробная формулировка задания`**

In [60]:
# код выполнения задания

***`Подробный вывод по заданию, описание полученных результатов`***